### PR TITLE
Adds :IsDescendantOf

### PR DESCRIPTION
**Pull request by Kampfkarren**

### CODE REVIEW

File: lib/instances/BaseInstance_spec.lua [hunk 1]
```diff
@@ -351,6 +351,30 @@ describe("instances.BaseInstance", function()
...
+			assert.False(BaseInstance:new():IsDescendantOf(parent))
+
+			local someOtherParent = BaseInstance:new()
+			local childOfSomethingElse = BaseInstance:new()
+			childOfSomethingElse.Parent = someOtherParent
+			assert.False(childOfSomethingElse:IsDescendantOf(parent))
+
+			local descendantInSomethingElse = BaseInstance:new()
+			descendantInSomethingElse.Parent = child
+			assert.True(descendantInSomethingElse:IsDescendantOf(parent))
```
Comment (LPGhatguy) [inline lib/instances/BaseInstance_spec.lua]: Can we break these up into separate tests?

File: lib/instances/BaseInstance.lua [hunk 2]
```diff
@@ -220,6 +220,16 @@ function BaseInstance.prototype:IsA(className)
...
 
+function BaseInstance.prototype:IsDescendantOf(object)
+	local parent = self
+
+	repeat
+		parent = parent.Parent
+	until parent == nil or parent == object
+
+	return parent == object
+end
```
Comment (LPGhatguy) [inline lib/instances/BaseInstance.lua]: This is pretty elegant!

File: lib/instances/BaseInstance_spec.lua (hunk 1 above)
Comment (Kampfkarren) [reply lib/instances/BaseInstance_spec.lua]: Will do.

**Comment by Kampfkarren** [comment]
Fixd.

**Comment by LPGhatguy** [comment]
Thanks!

**Pull request merged**